### ISSUE TITLE
Double initialization of a component

### ISSUE THREAD

**Issue by emakarov** [vercel/next.js#36438]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
npx --no-install next info

    Operating System:
      Platform: linux
      Arch: x64
      Version: #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022
    Binaries:
      Node: 16.14.2
      npm: 8.5.0
      Yarn: 1.22.5
      pnpm: N/A
    Relevant packages:
      next: 12.1.5
      react: 18.0.0
      react-dom: 18.0.0

warn  - Latest canary version not detected, detected: "12.1.5", newest: "12.1.6-canary.6".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
  ```

### What browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

npm run dev

### Describe the Bug

I have the simplest index.js webpage, which looks like this:
```
import { useEffect } from 'react';

export default function Home() {
  useEffect(() => {
    console.log('LOADED')
  }, [])
}
```

And I expect that `LOADED` is printed only once in a console, but I see it printed twice.

### Expected Behavior

Component should be mounted only once, i.e. I should see `LOADED` only once

### To Reproduce

Follow the installation procedure and replace the `pages/index.js` qith the snipped from description

```
npx create-next-app@latest
# replace index.js with following code:

import { useEffect } from 'react';

export default function Home() {
  useEffect(() => {
    console.log('LOADED')
  }, [])
}

# run the npm run 
dev and open chrome dev console
```
<img width="609" alt="Screenshot 2022-04-25 at 3 48 04 PM" src="https://user-images.githubusercontent.com/395963/165084284-29f1dee7-e884-4aa2-973d-26c7281f9920.png">

**Comment by balazsorban44** [comment]
This is working as intended, please have a look at https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-strict-mode
